tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php

```diff
@@ -380,6 +380,49 @@ public function checkInvoiceShippingTaxDetails(string $invoicePosition, string $
         }
     }
 
+    /**
+     * @Then the :invoicePosition invoice from order :orderReference should have following product tax details:
+     *
+     * @param string $invoicePosition
+     * @param string $orderReference
+     * @param TableNode $table
+     */
+    public function checkInvoiceProductTaxDetails(string $invoicePosition, string $orderReference, TableNode $table)
+    {
+        $orderId = SharedStorage::getStorage()->get($orderReference);
+        $invoiceProductData = $table->getColumnsHash();
+
+        $order = new Order($orderId);
```

Comment:
no need to cast ?

Reply:
answer myself :D :D https://github.com/PrestaShop/PrestaShop/pull/22018/files/b8f12216bcab2f3371f8a2cbe92ecdfed8531e6c#diff-69d56aaccff09b97252556f22677ccb8fcb8e8e67c1bfcff11981584723c3c61L164